cpp/include/cugraph/edge_properties.hpp

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cugraph/edge_property.hpp>
+#include <cugraph/utilities/error.hpp>
+
+#include <raft/core/device_span.hpp>
+
+#include <rmm/device_uvector.hpp>
+
+#include <variant>
+
+namespace cugraph {
+
+using numeric_device_uvector_t    = std::variant<std::monostate,
+                                                 rmm::device_uvector<float>,
+                                                 rmm::device_uvector<double>,
+                                                 rmm::device_uvector<int32_t>,
+                                                 rmm::device_uvector<int64_t>,
+                                                 rmm::device_uvector<size_t>>;
+using numeric_device_span_t       = std::variant<std::monostate,
+                                                 raft::device_span<float>,
+                                                 raft::device_span<double>,
+                                                 raft::device_span<int32_t>,
+                                                 raft::device_span<int64_t>,
+                                                 raft::device_span<size_t>>;
+using const_numeric_device_span_t = std::variant<std::monostate,
+                                                 raft::device_span<float const>,
+                                                 raft::device_span<double const>,
+                                                 raft::device_span<int32_t const>,
+                                                 raft::device_span<int64_t const>,
+                                                 raft::device_span<size_t const>>;
+
+template <typename edge_t>
+using edge_property_views_t = std::variant<std::monostate,
+                                           cugraph::edge_property_view_t<edge_t, float const*>,
+                                           cugraph::edge_property_view_t<edge_t, double const*>,
+                                           cugraph::edge_property_view_t<edge_t, int32_t const*>,
+                                           cugraph::edge_property_view_t<edge_t, int64_t const*>,
+                                           cugraph::edge_property_view_t<edge_t, size_t const*>>;
```

Comment:
Add edge_property_mutable_view_t for completeness?

And why `_views_t` instead of `_view_t` here?

Reply:
And for consistency in naming, shouldn't this better be `edge_arithmetic_property_veiw_t`?